src/components/CounterInput/index.js

```diff
@@ -0,0 +1,244 @@
+import React, { useRef, useImperativeHandle } from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier, useReduxForm, useErrorMessageId, useLabelId } from '../../libs/hooks';
+import RenderIf from '../RenderIf';
+import RelativeElement from '../Structural/relativeElement';
+import Label from '../Input/label';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import PlusIcon from './icons/plus';
+import MinusIcon from './icons/minus';
+import getNormalizedValue from './helpers/getNormalizedValue';
+import getValueOfNan from './helpers/getValueOfNan';
+import isButtonDisabled from './helpers/isButtonDisabled';
+import isMax from './helpers/isMax';
+import isMin from './helpers/isMin';
+import { StyledContainer, StyledInput, StyledButton, ButtonContainer } from './styled';
+
+/**
+ * CounterInput allows to increase or decrease a numerical value.
+ * @category Form
+ */
+
+const CounterInput = React.forwardRef((props, ref) => {
+    const {
+        id,
+        name,
+        value,
+        placeholder,
+        onChange,
+        tabIndex,
+        onFocus,
+        onBlur,
+        onClick,
+        onKeyDown,
+        disabled,
+        readOnly,
+        required,
+        min,
+        max,
+        isBare,
+        error,
+        variant,
+        step,
+        label,
+        labelAlignment,
+        hideLabel,
+        bottomHelpText,
+        className,
+        style,
+    } = useReduxForm(props);
+
+    const inputRef = useRef(null);
+    const inputId = useUniqueIdentifier('counter-input');
+    const errorMessageId = useErrorMessageId(error);
+    const labelId = useLabelId(label);
+    const isReadOnly = !!(!disabled && readOnly);
+    const finalStep = step || 1;
+
+    useImperativeHandle(ref, () => ({
+        focus: () => {
+            inputRef.current.focus();
+        },
+        click: () => {
+            inputRef.current.click();
+        },
+        blur: () => {
+            inputRef.current.blur();
+        },
+    }));
+
+    const handlePlusMouseDown = event => {
+        event.preventDefault();
+        inputRef.current.focus();
+        onChange(getNormalizedValue(Number(inputRef.current.value) + finalStep));
```

Comment:
I think we should use the prop `value` instead of `inputRef.current.value` since the source of truth is the prop `value`